Lib/test/test_unittest/test_runner.py

```diff
@@ -536,6 +536,44 @@ def testNothing(self):
 
         self.assertEqual(TestableTest._class_cleanups, [])
 
+    def test_run_nested_test(self):
+        ordering = []
+
+        class InnerTest(unittest.TestCase):
+            @classmethod
+            def setUpClass(cls):
+                ordering.append('setUpClass2')
```

Comment:
It might be clearer to change the string to "setup inner" (and likewise for the other numbered strings), that makes it clearer what the expected ordering is.